### DISCUSSION TITLE
Warning, C++ compiler Arm architecture

### DISCUSSION THREAD

**Discussion by danielbuubble** [microsoft/vscode-cpptools#10044] [General]
Hi!

I am taking a course at my university and I am pretty new to C++. I have been stuck on this "bug" where, whenever I press the play button to compile and run the file, I get this warning "default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]". And when I look at the debug console I get this message "Warning: Debuggee TargetArchitecture not detected, assuming x86_64.". From my limited knowledge, it looks like vscode assumes I am using arch x86, but I have a M2 arch Arm. My question is do you guys have any fix to my problem? Because I can run the code using commands in the terminal, but I would like the play button to work (Invested to much time in this to give up on the play button). I have looked all over the internett, but the solutions are way over my knowledge, and to advanced for me, I think. 
<img width="1470" alt="Skjermbilde 2022-10-21 kl  15 14 35" src="https://user-images.githubusercontent.com/116361422/197204793-114db8ff-b000-4873-9377-d5c7ac3eabf9.png">

Thank you for your help, and sorry for my limited knowledge and mispokenes. :D

**Comment by sean-mcmanus** [comment]
"default member initializer for non-static data member is a C++11 extension" is a clang or gcc warning, not from our extension. It sounds like you need to compile with -std=c++11 or higher (e.g. c++17, c++20).

I think "Warning: Debuggee TargetArchitecture not detected, assuming x86_64" can be ignored.

Why are these warnings blocking you?

**Reply by danielbuubble** [reply]
How do I compile with c++11 or higher when using the vscode run and debug play button?

The reasons these warning are blocking me, is when I run a test file, the test file won’t run. And I have used to much of my time to find out why the run and debug play button in vscode compiles my files with a version before C++11 to give up

**Reply by sean-mcmanus** [reply]
Open the .vscode/tasks.json file and add `"-std=c++11",` to the args array.

**Reply by sean-mcmanus** [reply]
We have a bug tracking improving the experience at https://github.com/microsoft/vscode-cpptools/issues/9600 . The user in that thread hit the same issue you did.

**Reply by danielbuubble** [reply]
That helped, the warning are now gone. But when I try to run the code, I still get the message above, that I am using the wrong architecture.

**Reply by sean-mcmanus** [reply]
Ignore that warning. I get it too on M1 Mac. I can still debug. If you're not able to debug, it may be caused by something else.
